rust_src/src/buffers.rs

```diff
@@ -1014,6 +1017,51 @@ pub fn erase_buffer() {
     }
 }
 
+/// Return a string that is the name of no existing buffer based on NAME.
+/// If there is no live buffer named NAME, then return NAME.
+/// Otherwise modify name by appending `<NUMBER>', incrementing NUMBER
+/// (starting at 2) until an unused name is found, and then return that name.
+/// Optional second argument IGNORE specifies a name that is okay to use (if
+/// it is in the sequence to be tried) even if a buffer with that name exists.
+///
+/// If NAME begins with a space (i.e., a buffer that is not normally
+/// visible to users), then if buffer NAME already exists a random number
+/// is first appended to NAME, to speed up finding a non-existent buffer.
+#[lisp_fn(min = "1")]
+pub fn generate_new_buffer_name(name: LispStringRef, ignore: LispObject) -> LispStringRef {
```

Comment:
The ignore argument can be an `Option<LispStringRef>`

Reply:
`ignore` is used only as an argument to `string_equal`. If I make `ignore` an `Option<LispStringRef>`, I have to have `.into()` calls. I decided making it a `LispObject` would be simpler.

Reply:
Yeah, @brotzeit had a similar thing on one of their PRs recently. Sometimes letting the `LispObject` through is the right thing.

My take on it is whether the value is used in the Rust function or just passed on through. If Rust touches, make a Rust type and then use `.into()`. Otherwise, just leave it a `LispObject`.

Reply:
Also, `ignore` here can be a symbol or a string. So it would be wrong to make it an `Option<LispStringRef>`.